index.js

```diff
@@ -27,15 +25,23 @@ function underPressure (fastify, opts, next) {
   timer.unref()
 
   var externalsHealthy = false
-  const doCheck = () => healthCheck()
-    .then(externalHealth => { externalsHealthy = externalHealth })
-    .catch((error) => {
-      externalsHealthy = false
-      fastify.log.error('external healthCheck function suupplied to `under-pressure` threw an error. setting the service status to unhealthy.', { error })
-    })
-  doCheck()
-  const externalHealthCheckTimer = setInterval(doCheck, healthCheckInterval)
-  externalHealthCheckTimer.unref()
+  var externalHealthCheckTimer
+  if (healthCheck) {
+    assert(typeof healthCheck === 'function', 'opts.healthCheck should be a function that returns a promise that resolves to true or false')
+
+    const doCheck = () => healthCheck()
+      .then(externalHealth => { externalsHealthy = externalHealth })
+      .catch((error) => {
+        externalsHealthy = false
+        fastify.log.error('external healthCheck function suupplied to `under-pressure` threw an error. setting the service status to unhealthy.', { error })
+      })
+    doCheck()
```

Comment:
we need to wait the first run of this before calling `next`.

Reply:
I've added a fix for this, but dislike the code I created because of it. The if  guard around the next() call at the end of the function seems out of place. I've added a comment there but other thoughts/feedback needed.